src/state/groups/actions.js

```diff
@@ -1,16 +1,26 @@
-import { getApiGroups } from 'api/groups';
-import { ADD_GROUPS } from 'state/groups/types';
+import { SET_GROUPS } from 'state/groups/types';
+import { getGroups } from 'api/groups';
+import { setPage } from 'state/pagination/actions';
 
-export const addGroups = groups => ({
-  type: ADD_GROUPS,
+export const setGroups = groups => ({
+  type: SET_GROUPS,
   payload: {
     groups,
   },
 });
 
 // thunk
-export const fetchGroups = () => dispatch =>
-  getApiGroups().then((data) => {
-    dispatch(addGroups(data.payload));
+export const fetchGroups = (page = { page: 1, pageSize: 10 }, params = '') => dispatch =>
+  new Promise((resolve) => {
+    getGroups(page, params).then((response) => {
```

Comment:
What if the promise get rejected? The Promise would remain pending forever.
You need to add `.catch()` blocks in internal promises, something like

```javascript
export const fetchGroups = (page = { page: 1, pageSize: 10 }, params = '') => dispatch =>
  new Promise((resolve, reject) => {
    getGroups(page, params).then((response) => {
      if (response.ok) {
        response.json().then((data) => {
          dispatch(setGroups(data.payload));
          dispatch(setPage(data.metaData));
          resolve();
        })
        .catch(reject);
      } else {
        resolve();
      }
    })
    .catch(reject);
  });
```